prog/size_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2016 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package prog
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestAssignSizeRandom(t *testing.T) {
+	rs, iters := initTest(t)
+	for i := 0; i < iters; i++ {
+		p := Generate(rs, 10, nil)
+		for _, call := range p.Calls {
+			assignSizesCall(call)
+			assignTypeAndDir(call)
+		}
+		for try := 0; try <= 10; try++ {
+			p.Mutate(rs, 10, nil)
+			for _, call := range p.Calls {
+				assignSizesCall(call)
```

Comment:
Check that assignSizesCall does not change program here as well.
We may forget to update sizes on some paths in Mutate. The check will catch that.